src/psyclone/alg_gen.py

```diff
@@ -40,21 +40,18 @@
 '''
 
 from __future__ import absolute_import
+from psyclone.errors import PsycloneError
```

Comment:
For consistency with other class names, please could you make this "PSycloneError" rather than "PsycloneError".